src/risk_distributions/risk_distributions.py

```diff
@@ -469,53 +469,19 @@ def get_parameters(cls, weights: Parameters,
 
         return weights, params
 
-    def pdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
-        single_val = isinstance(x, (float, int))
-        values_only = isinstance(x, np.ndarray)
-
-        x, weights = format_call_data(x, self.weights)
-
-        computable = weights[(weights.sum(axis=1) != 0) & ~np.isnan(x)].index
-
-        p = pd.Series(np.nan, index=x.index)
-
-        if not computable.empty:
-            p.loc[computable] = 0
-            for name, parameters in self.parameters.items():
-                w = weights.loc[computable, name]
-                params = parameters.loc[computable] if len(parameters) > 1 else parameters
-                p += w * self._distribution_map[name](parameters=params).pdf(x.loc[computable])
-
-        if single_val:
-            p = p.iloc[0]
-        if values_only:
-            p = p.values
-        return p
 
-    def ppf(self, q: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
-        single_val = isinstance(q, (float, int))
-        values_only = isinstance(q, np.ndarray)
-
-        q, weights = format_call_data(q, self.weights)
+    def pdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
+        return self._compute('pdf', x)
 
-        computable = weights[(weights.sum(axis=1) != 0) & ~np.isnan(q)].index
 
-        x = pd.Series(np.nan, index=q.index)
+    def cdf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
+        return self._compute('cdf', x)
 
-        if not computable.empty:
-            x.loc[computable] = 0
-            for name, parameters in self.parameters.items():
-                w = weights.loc[computable, name]
-                params = parameters.loc[computable] if len(parameters) > 1 else parameters
-                x += w * self._distribution_map[name](parameters=params).ppf(q.loc[computable])
 
-        if single_val:
-            x = x.iloc[0]
-        if values_only:
-            x = x.values
-        return x
+    def ppf(self, x: Union[pd.Series, np.ndarray, float, int]) -> Union[pd.Series, np.ndarray, float]:
```

Comment:
Several things:

- The arg here should still be `q`.  The pdf and cdf functions take an exposure and return a density or a probability associated with that exposure, respectively.  The ppf function takes a probability (or quantile or propensity) and finds the associated exposure.
- The implementation we discussed at this level of abstraction takes two propensity arguments, say `q`, which get's used in the call to the underlying distribution, and `q_w` which gets used to select the distribution.